feature-libs/cart/import-export/components/export-entries/export-entries.service.ts

```diff
@@ -41,4 +82,33 @@ export class ExportEntriesService {
       filter((entries) => entries?.length > 0)
     );
   }
+
+  exportEntries() {
+    const names: string[] = [];
+    const values: any[] = [];
+
+    this.columns.map((column) => {
+      this.translationService
+        .translate(`exportEntries.columnNames.${column.name.key}`)
+        .pipe(take(1))
+        .subscribe((name) => names.push(name));
+    });
+
+    this.getEntries()
+      .pipe(take(1))
+      .subscribe((entries) => {
+        entries.map((entry) => {
+          values.push(
+            Object.assign(
+              {},
+              this.columns.map((column) =>
+                this.resolveValue(column.value, entry)
+              )
+            )
+          );
+        });
+      });
+
+    return [{ ...names }, ...values];
```

Comment:
I don't know why it works, but it looks like strong anti-pattern. We shouldn't use subscribe inside synchronous method and return obtained value, we can't count on subscription to trigger faster than method termination. Good luck with unit testing that 😅

Reply:
As I need those changes in epic for now I created follow-up ticket for this one: #13007